doc/source/ray-job-submission/overview.rst

```diff
@@ -0,0 +1,228 @@
+============================
+From laptop to production
+============================
+
+**Ray Job Submission**
+
+**Concepts**
+
+- **Package**: A collection of files and configurations that defines an application thus allows it to be executed in a different environment remotely (ideally self-contained).
+
+- **Job**: A Ray application that can successfully execute on a user laptop, also can be submitted to a remote entity with specs such that its execution and lifetime is beyond a single ray cluster.
+Open source Ray will have the functionality to run jobs, but not manage them outside of a cluster.
+
+- **Job Manager**: An entity that manages the lifecycle of a Job and potentially multiple ray clusters, such as scheduling, killing, polling status, getting logs, and persisting inputs / outputs.
+Should be highly available (HA) by default
+Can be any framework with these abilities, such as Airflow.
+
+
+**The goal of Ray job submssion is to provide a lightweight mechanism for user to submit their locally developed and tested application to a running local / remote ray cluster, thus enable user to package, deploy, and manage their ray application as jobs, with pluggable job manager of their choice.**
+
+**Example - Setup**
+
+Let's start with writing a local driver script to a temporary directory and use it as example for job submission. Once executed locally, this script will use ray APIs to print counter value of a remote actor from 1 to 5, and print the version of 'requests' module it's using.
+
+.. tabs::
+  .. group-tab:: Python
+
+.. code-block:: python
+    import os
+    import tempfile
+    import time
+
+    import ray
+
+    driver_script = """
+import ray
+import requests
+
+ray.shutdown()
+ray.init()
+
+@ray.remote
+class Counter:
+    def __init__(self):
+        self.counter = 0
+
+    def inc(self):
+        self.counter += 1
+
+    def get_counter(self):
+        return self.counter
+
+counter = Counter.remote()
+
+for _ in range(5):
+    ray.get(counter.inc.remote())
+    print(ray.get(counter.get_counter.remote()))
+
+print(requests.__version__)
+    """
+
+    # Flag to enable runtime_env in case your local ray is built from source
+    os.environ["RAY_RUNTIME_ENV_LOCAL_DEV_MODE"] = "1"
```

Comment:
this is an internal developer flag, don't need it here?

Reply:
This is what @architkulkarni suggested me to set if im testing runtime_env with a wheel built on master for per-actor runtime_env, so I put it there in case the user is a ray developer that doesn't use public release

Reply:
let's not expose it to users in docs, this is a rare situation and it prints a warning when it happens